plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterDestroyWorker.java

```diff
@@ -258,6 +296,14 @@ public boolean destroy() throws CloudRuntimeException {
                     updateKubernetesClusterEntryForGC();
                     throw new CloudRuntimeException(msg, e);
                 }
+                try {
+                    releaseVpcTierPublicIpIfNeeded();
+                } catch (InsufficientAddressCapacityException e) {
```

Comment:
Not sure if Insufficient address capacity would be the most appropriate exception type?

Reply:
@Pearl1594 that is what NetworkService.releaseIpAddress throws

Reply:
Got It @shwstppr. Thanks